### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by joshgubler**
Replace CodePipeline with GitHub Actions.

We probably don't want to merge this until we're ready for the whole org to be using GitHub Actions (i.e. until we have secrets for all the AWS accounts).

### CODE REVIEW

**Comment by jvisker** [comment]
The node modules make this code review more difficult...

**Comment by joshgubler** [comment]
That's fair. Not sure what to do about that. We can wait until that action gets moved out to its own repo.

**Comment by GaryGSC** [comment]
I can fix that. One sec.

**Comment by GaryGSC** [comment]
Alright. I believe that should work.

File: terraform-iac/dev/app/main.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 terraform {
   backend "s3" {
-    bucket         = "terraform-state-storage-<account_number>"
-    dynamodb_table = "terraform-state-lock-<account_number>"
+    bucket         = "terraform-state-storage-977306314792"
+    dynamodb_table = "terraform-state-lock-977306314792"
```
Comment (GaryGSC) [inline terraform-iac/dev/app/main.tf:4]: 👀

File: terraform-iac/dev/setup/setup.tf [hunk 2]
```diff
@@ -1,7 +1,7 @@
 terraform {
   backend "s3" {
-    bucket         = "terraform-state-storage-<account_number>"
-    dynamodb_table = "terraform-state-lock-<account_number>"
+    bucket         = "terraform-state-storage-977306314792"
+    dynamodb_table = "terraform-state-lock-977306314792"
```
Comment (GaryGSC) [inline terraform-iac/dev/setup/setup.tf:4]: 👀

File: terraform-iac/dev/app/main.tf (hunk 1 above)
Comment (joshgubler) [reply terraform-iac/dev/app/main.tf:4]: Good catch. I'll change that back, but it'll break the pipeline.

Comment (joshgubler) [reply terraform-iac/dev/app/main.tf]: done

File: terraform-iac/dev/setup/setup.tf (hunk 2 above)
Comment (joshgubler) [reply terraform-iac/dev/setup/setup.tf]: done

**Review by GaryGSC — DISMISSED** [review]

**Comment by joshgubler** [comment]
Now that we have secrets ready for all of the accounts, I think we're ready to merge this to master.

File: .github/workflows/pipeline-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  build_and_deploy:
+    name: Build and deploy Fargate API
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Setup DEV Environment Variables
+      if: github.ref == 'refs/heads/dev'
+      run: |
+        echo "::set-env name=aws_key::${{ secrets.byu_oit_terraform_dev_key }}"
+        echo "::set-env name=aws_secret::${{ secrets.byu_oit_terraform_dev_secret }}"
```
Comment (jvisker) [inline .github/workflows/pipeline-workflow.yml:20]: Do we need to indicate here that some of these secrets need to be setup by the developer?

File: terraform-iac/prd/app/main.tf [hunk 4]
```diff
@@ -20,8 +20,21 @@ module "app" {
   source    = "../../modules/app/"
   env       = "prd"
   image_tag = var.image_tag
+  codedeploy_termination_wait_time = 15
```
Comment (jvisker) [inline terraform-iac/prd/app/main.tf:23]: I thought we wanted this shorter than 15 minutes

**Review by jvisker — CHANGES_REQUESTED** [review]

File: terraform-iac/prd/app/main.tf (hunk 4 above)
Comment (joshgubler) [reply terraform-iac/prd/app/main.tf:23]: 0 in dev. 15 in prd

File: .github/workflows/pipeline-workflow.yml (hunk 3 above)
Comment (joshgubler) [reply .github/workflows/pipeline-workflow.yml:20]: I considered that. There are lots of pieces that need to be customized when this template is instantiated. I think I captured them all in README.md

**Review by jvisker — APPROVED** [review]

File: .github/workflows/pipeline-workflow.yml (hunk 3 above)
Comment (joshgubler) [reply .github/workflows/pipeline-workflow.yml:20]: Or maybe I didn't do a great job with that.

Comment (jvisker) [reply .github/workflows/pipeline-workflow.yml:20]: I wasn't really looking at it from the readme perspective, but I'm the kind of guy that never reads the instructions on things either.

**Pull request merged**